modules/ROOT/pages/logging-in-mule.adoc

```diff
@@ -0,0 +1,287 @@
+= Configuring Logging
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: mule, studio, logger, logs, log, notifications, errors, debug
+
+Mule automatically logs multiple messages and specific elements in your application flows to help you debug and keep track of events. You can also include the xref:logger-component-reference.adoc[`<logger>` Element] anywhere in a flow and set it up to output any message you want.
+
+You can create a configuration file to define:
+
+ * What kinds of messages to log
+ * How to log the messages (asynchronously or synchronously)
+ * Where to log the messages, such as to the console or disk, or to an endpoint or database.
+
+Mule uses http:ç, which is a logging facade that discovers and uses a logging strategy from the classpath, such as Apache Log4j 2 or the JDK Logger. By default, Mule includes Log4j 2, which is configured with a file called `log4j2.xml`.
+
+The Mule server has a `log4j2.xml` in its `conf` directory, which you can customize when running the server in standalone mode. This configuration file is relevant for data output to the `mule_ee.log` file.
+
+If you remove the `çlog4j2.xml` file from your application's classpath, Mule Runtime applies a default logging configuration to the application and outputs the log to `MULE_HOME/logs/<application-name>.log`.
+
+== Viewing Logs
+To view Anypoint Studio logs:
+
+. Click *Anypoint Studio* > *About Anypoint Studio* > *Installation Details* > *Configuration* tab > *View Error Log*.
++
+image::logging-in-mule-view-error-log.png[View Error Log]
+
+The output from the View Error Log is in the workspace's `.metadata/.log` file, for example:
+----
+/Users/me/AnypointStudio/workspace/.metadata/.log
+----
+
+After you run your Mule application, the log output is in the workspace's `.mule/logs/<project>.log` file.
+For example, for project `abcd`:
```

Comment:
I don't get a logs/ subdir under the .mule dir in my workspace whether I'm running my app or not:
```
sduke-ltm:.mule sduke$ pwd
/Users/sduke/AnypointStudio/studio-workspace7/.mule
sduke-ltm:.mule sduke$ ls -la
total 0
drwxr-xr-x   3 sduke  wheel   96 Feb 11 19:58 .
drwxr-xr-x  28 sduke  wheel  896 Mar 18 17:11 ..
drwxr-xr-x  21 sduke  wheel  672 Mar  8 19:17 plugins-tmp
```
The only .log files I see in my workspace are these:
```
sduke-ltm:studio-workspace7 sduke$ find ./ -name *.log
.//.metadata/.bak_0.log
.//.metadata/.log
```

Reply:
Alejandro, can you comment on this?

Reply:
/Applications/AnypointStudio.app/Contents/Eclipse/plugins/org.mule.tooling.server.4.1.4.ee_7.3.2.201902011858/mule/logs/{appName}.log
Logs should be inside that folder.

Reply:
This file is for all of the Anypoint Studio logs for all Studio versions that run Mule 4? I am seeing a slightly different file on my workspace but am not seeing a log file (maybe because I haven't turned it on):

 org.mule.tooling.server.4.1.4.ee_7.3.0.201810161635

Reply:
I think this is wrong, logs are not inside the `.mule` folder, but in `logs/{appName}.log`